test/plugin/test_filter_add_insert_ids.rb

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative '../helper'
+
+require 'fluent/test/driver/filter'
+require 'fluent/plugin/filter_add_insert_ids'
+
+# Unit tests for filter_add_insert_ids plugin.
+class FilterAddInsertIdsTest < Test::Unit::TestCase
+  include Fluent::Plugin::AddInsertIdsFilter::ConfigConstants
+
+  CUSTOM_INSERT_ID_KEY = 'custom_insert_id_key'.freeze
+  INSERT_ID = 'aeyr82r92h249gh9h'.freeze
+  TEST_MESSAGE = 'test message for add_insert_ids plugin.'.freeze
+  APPLICATION_DEFAULT_CONFIG = ''.freeze
+  INSERT_ID_KEY_CONFIG = %(
+    insert_id_key #{CUSTOM_INSERT_ID_KEY}
+  ).freeze
+
+  def setup
+    Fluent::Test.setup
+  end
+
+  def test_configure_insert_id_key
+    {
+      APPLICATION_DEFAULT_CONFIG => DEFAULT_INSERT_ID_KEY,
+      INSERT_ID_KEY_CONFIG => CUSTOM_INSERT_ID_KEY
+    }.each do |config, insert_id_key|
+      d = create_driver(config)
+      assert_equal insert_id_key, d.instance.insert_id_key
+    end
+  end
+
+  def test_add_insert_ids
+    total_entry_count = 1000
```

Comment:
Could this number be smaller (like 5)? Or do we need to test with a high volume?

Reply:
The number was chose to be this high mainly to test against any potential race condition (there should not be).